bundles/org.openhab.binding.roku/src/main/java/org/openhab/binding/roku/internal/handler/RokuHandler.java

```diff
@@ -0,0 +1,241 @@
+            thing.setProperty(PROPERTY_MODEL_NUMBER, deviceInfo.getModelNumber());
+            thing.setProperty(PROPERTY_DEVICE_LOCAITON, deviceInfo.getUserDeviceLocation());
+            thing.setProperty(PROPERTY_SERIAL_NUMBER, deviceInfo.getSerialNumber());
+            thing.setProperty(PROPERTY_DEVICE_ID, deviceInfo.getDeviceId());
+            thing.setProperty(PROPERTY_SOFTWARE_VERSION, deviceInfo.getSoftwareVersion());
+            updateStatus(ThingStatus.ONLINE);
+        } catch (RokuHttpException e) {
+            logger.debug("Unable to retrieve Roku device-info. Exception: {}", e.getMessage(), e);
+        }
+        startAutomaticRefresh();
+        startAppListRefresh();
+    }
+
+    /**
+     * Start the job to periodically get status updates from the Roku
+     */
+    private void startAutomaticRefresh() {
+        ScheduledFuture<?> refreshJob = this.refreshJob;
+        if (refreshJob == null || refreshJob.isCancelled()) {
+            Runnable runnable = () -> {
+                synchronized (sequenceLock) {
+                    try {
+                        ActiveApp activeApp = communicator.getActiveApp();
+                        updateState(ACTIVE_APP, new StringType(activeApp.getApp().getId()));
+                        updateStatus(ThingStatus.ONLINE);
+                    } catch (RokuHttpException e) {
+                        logger.debug("Unable to retrieve Roku active-app info. Exception: {}", e.getMessage(), e);
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                    }
+
+                    try {
+                        Player playerInfo = communicator.getPlayerInfo();
+                        // When nothing playing, 'close' is reported, replace with 'stop'
+                        updateState(PLAY_MODE, new StringType(playerInfo.getState().replaceAll(CLOSE, STOP)));
+
+                        // Remove non-numeric from string, ie: ' ms'
+                        String position = playerInfo.getPosition().replaceAll(NON_DIGIT_PATTERN, EMPTY);
+                        if (!EMPTY.equals(position)) {
+                            updateState(TIME_ELAPSED,
+                                    new QuantityType<>(Integer.parseInt(position) / 1000, API_SECONDS_UNIT));
+                        } else {
+                            updateState(TIME_ELAPSED, UnDefType.UNDEF);
+                        }
+
+                        String duration = playerInfo.getDuration().replaceAll(NON_DIGIT_PATTERN, EMPTY);
+                        if (!EMPTY.equals(duration)) {
+                            updateState(TIME_TOTAL,
+                                    new QuantityType<>(Integer.parseInt(duration) / 1000, API_SECONDS_UNIT));
+                        } else {
+                            updateState(TIME_TOTAL, UnDefType.UNDEF);
+                        }
+                    } catch (RokuHttpException | NumberFormatException e) {
```

Comment:
By having 1 exception for all you might miss updating channels. Because if it fails on the first channel set, the other channels are not set. Also if only 1 fails on `NumberFormatException` should the thing go offline?

Reply:
I looked at this again and removed the NumberFormatException from the catch. The code already removes non-numeric characters and checks for empty string before doing parseInt(). Catching NumberFormatException was redundant.